From ba871510ef6d3a3223be54d0aa81c9f9797ba103 Mon Sep 17 00:00:00 2001
From: Tianling Shen <cnsztl@immortalwrt.org>
Date: Sat, 15 Jul 2023 17:31:44 +0800
Subject: [PATCH] rockchip: add NanoPi R6C support

Hardware
--------
RockChip RK3588S ARM64 (8 cores)
4/8GB LPDDR4X RAM
1000 Base-T (native, rtl8211f)
2500 Base-T (PCIe, rtl8125b)
4 LEDs (SYS / WAN / LAN / USER)
2 Button (GPIO Reset, MaskROM)
32GB eMMC on-board
Micro-SD Slot
USB 3.0 Port
USB 2.0 Port
HDMI 2.1
USB Type C PD 5/9/12/20V

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>
---
 .../armv8/base-files/etc/board.d/01_leds      |   1 +
 .../armv8/base-files/etc/board.d/02_network   |   2 +
 .../etc/hotplug.d/net/40-net-smp-affinity     |   1 +
 .../boot/dts/rockchip/rk3588s-nanopi-r6c.dts  |  27 +++
 .../boot/dts/rockchip/rk3588s-nanopi-r6s.dts  | 185 ++++++++++++++++-
 target/linux/rockchip/image/armv8.mk          |   9 +
 ...-rockchip-Add-RK806-regulator-config.patch | 196 ------------------
 ...ockchip-Add-missing-regulator-config.patch |  21 --
 .../210-rockchip-rk356x-add-support-for-new-boards.patch        |   3 +-
 9 files changed, 226 insertions(+), 219 deletions(-)
 create mode 100644 target/linux/rockchip/files/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6c.dts

diff --git a/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds b/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
index 92abc55f24a..df56abdd735 100644
--- a/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
+++ b/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
@@ -14,6 +14,7 @@ friendlyarm,nanopi-r2s|\
 friendlyarm,nanopi-r4s|\
 friendlyarm,nanopi-r4se|\
 friendlyarm,nanopi-r4s-enterprise|\
+friendlyarm,nanopi-r6c|\
 xunlong,orangepi-r1-plus|\
 xunlong,orangepi-r1-plus-lts)
 	ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth0"
diff --git a/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity b/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity
index 23dc213dea2..d0b6cb39a60 100644
--- a/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity
+++ b/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity
@@ -32,6 +32,7 @@ case "$(board_name)" in
 ezpro,mrkaio-m68s|\
 firefly,rk3568-roc-pc|\
 friendlyarm,nanopi-r5c|\
+friendlyarm,nanopi-r6c|\
 lunzn,fastrhino-r66s)
 	set_interface_core 2 "eth0"
 	set_interface_core 4 "eth1"
